gradle-baseline-java/build.gradle

```diff
@@ -131,3 +131,21 @@ project.ext.'gradle.publish.secret' = System.env["GRADLE_SECRET"]
 tasks.withType(Test) {
     systemProperty 'recreate', System.getProperty('recreate', 'false')
 }
+
+configurations {
+    checkstyleVersion
+}
+
+dependencies {
+    checkstyleVersion 'com.puppycrawl.tools:checkstyle'
+}
+
+task writeCheckstyleVersion() {
+    doLast {
+        file('src/main/resources/checkstyle.version').text = getVersion('com.puppycrawl.tools', 'checkstyle', configurations.checkstyleVersion)
+    }
+}
+
+if (gradle.startParameter.isWriteDependencyLocks()) {
+    gradle.startParameter.taskNames += 'writeCheckstyleVersion'
+}
```

Comment:
an alternative approach here is to generate the resource just-in-time as a dependency of the ./gradlew idea task... no super strong opinions, i just kinda like that this one is reasonably bulletproof

Reply:
Also no strong opinion but I like doing this as part of "--write-locks" as people already expect their version numbers to change through this run.

Reply:
yeah, in general I hate it when i open intellij and something crucial is missing because it failed to generate somehow.  this seemed more ~idiot~ dan-proof